specification/extendedlocation/resource-manager/Microsoft.ExtendedLocation/preview/2021-03-15-preview/customlocations.json

```diff
@@ -0,0 +1,708 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/EnabledResourceTypesListResult"
+            }
+          },
+          "400": {
+            "description": "Bad Request",
+            "x-ms-error-response": true
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Get Custom Location": {
+            "$ref": "./examples/CustomLocationsListEnabledResourceTypes.json"
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "EnabledResourceTypeNameParameter": {
+      "name": "enabledResourceTypeName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "minLength": 1,
+      "maxLength": 63,
+      "pattern": "^[a-zA-Z0-9]$|^[a-zA-Z0-9][-_a-zA-Z0-9]{0,61}[a-zA-Z0-9]$",
+      "description": "Custom Locations Enabled Resource Type name."
+    },
+    "ResourceNameParameter": {
+      "name": "resourceName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "minLength": 1,
+      "maxLength": 63,
+      "pattern": "^[a-zA-Z0-9]$|^[a-zA-Z0-9][-_a-zA-Z0-9]{0,61}[a-zA-Z0-9]$",
+      "description": "Custom Locations name."
+    }
+  },
+  "definitions": {
+    "EnabledResourceType": {
+      "description": "EnabledResourceType definition.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/EnabledResourceTypeProperties",
+          "description": "The set of properties for EnabledResourceType specific to a Custom Location",
+          "x-ms-client-flatten": true
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "description": "Metadata pertaining to creation and last modification of the resource",
+          "readOnly": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
```

Comment:
so there will be an id property on the enabled resource type response. What will the name segment in this id property value correspond to? Will it offer any value to the end customer?

Reply:
It sounds like this was a suggestion from ARM implemented before I started. The name segment is a sha256 hash of the cluster extension the enabledResoruceType belongs to. I will follow up.